test/unit/spec/signaling/v2/peerconnection.js

```diff
@@ -1326,84 +1346,96 @@ describe('PeerConnectionV2', () => {
-            type: 'icecandidate',
-            candidate: { candidate: 'candidate2' }
-          });
+        if (which === 'final') {
+          iceStatePromise = new Promise(resolve => test.pcv2.once('candidates', resolve));
+        }
 
-          if (which === 'final') {
-            iceStatePromise = new Promise(resolve => test.pcv2.once('candidates', resolve));
-          }
+        test.pc.emit('icecandidate', {
+          type: 'icecandidate',
+          candidate: null
+        });
 
-          test.pc.emit('icecandidate', {
-            type: 'icecandidate',
-            candidate: null
-          });
+        iceState = await iceStatePromise;
+      });
 
-          iceState = await iceStatePromise;
+      if (iceLite && which !== 'final') {
+        it('should not emit the event', () => {
+          assert.deepEqual(iceState.ice.candidates, []);
+          assert(iceState.ice.complete);
         });
+        return;
+      }
 
-        context('emits the event', () => {
-          it('with the correct ID', () => {
-            assert.equal(iceState.id, test.pcv2.id);
-          });
+      context('emits the event', () => {
+        it('with the correct ID', () => {
+          assert.equal(iceState.id, test.pcv2.id);
+        });
 
-          if (which === 'initial') {
-            it('with a single-element list of ICE candidates', () => {
-              assert.deepEqual(
-                iceState.ice.candidates,
-                [{ candidate: 'candidate1' }]);
-            });
-          } else {
-            it('with the full list of ICE candidates gathered up to that point', () => {
-              assert.deepEqual(
-                iceState.ice.candidates,
-                [{ candidate: 'candidate1' },
-                 { candidate: 'candidate2' }]);
-            });
-          }
+        if (which === 'initial') {
+          it('with a single-element list of ICE candidates', () => {
+            assert.deepEqual(
+              iceState.ice.candidates,
+              [{ candidate: 'candidate1' }]);
+          });
+        } else {
+          it(iceLite ? 'with no candidates' : 'with the full list of ICE candidates gathered up to that point', () => {
+            assert.deepEqual(iceState.ice.candidates, iceLite
+              ? [] : [{ candidate: 'candidate1' }, { candidate: 'candidate2' }]);
+          });
+        }
 
-          if (which === 'final') {
-            it('with completed set to true', () => {
-              assert(iceState.ice.complete);
-            });
-          } else {
-            it('with completed unset', () => {
-              assert(!iceState.ice.complete);
-            });
-          }
-        });
+        if (which === 'final') {
+          it('with completed set to true', () => {
+            assert(iceState.ice.complete);
+          });
+        } else {
+          it('with completed unset', () => {
+            assert(!iceState.ice.complete);
+          });
+        }
```

Comment:
I believe this test confirms that PeerConnection will emit a single event with no candidates and ice.complete set to true.